deepnote_core/resources/scripts/resource_usage.py

```
@@ -0,0 +1,256 @@
+
+    def log_message(self, format: str, *args: Any) -> None:
+        print(f"{self.address_string()} - {format % args}")
+
+
+def main() -> None:
+    port = int(os.environ.get("RESOURCE_USAGE_METRICS_PORT", 9104))
+    monitor = ResourceMonitor()
+    monitor.get_cpu()  # Initialize CPU baseline
+
+    handler = partial(MetricsHandler, monitor)
+    server = ThreadingHTTPServer(("0.0.0.0", port), handler)
```

Ruff (0.14.11)

Possible binding to all interfaces

(S104)